README.md

```diff
@@ -1,2 +1,296 @@
+
+A decorator enables you to provide metadata for Controller methods to set the minimum logLevel.
+
+### Example Usage
+```ts
+import {LogLevelMixin, LogComponent, LOG_LEVEL, log} from 'loopback4-example-log-extension';
+// Other imports ... 
+
+class LogApp extends LogLevelMixin(Application) {
+  constructor() {
+    super({
+      components: [RestComponent, LogComponent],
+      logLevel: LOG_LEVEL.ERROR,
+      controllers: [MyController]
+    });
+  };
+}
+
+class MyController {
+  @log(LOG_LEVEL.WARN)
+  @get('/')
+  hello() {
+    return 'Hello LoopBack';
+  }
+
+  @log(LOG_LEVEL.ERROR)
+  @get('/name')
+  helloName() {
+    return 'Hello Name'
+  }
+}
+```
+
+## Tutorial
+
+Start by cloning [loopback4-extension-starter]() and updating files headers to be `loopback4-example-log-extension` and `package.json`.
+
+Keep all the top level files but modify the `src` directory and `test` directory as follows.
+
+### `/src/index.ts`
+
+Update the exports accordingly here. Typically all files in `/src` are exported. Final version of file should be as follows:
+
+```ts
+export * from './decorators/log.decorator';
+export * from './mixins/log-level.mixin';
+export * from './providers/log-action.provider';
+export * from './providers/log-level.provider';
+export * from './providers/timer.provider';
+export * from './log.component';
+export * from './types';
+export * from './keys';
+```
+
+### `/src/keys.ts`
+
+Define new `Binding` values as well as exporting an `enum` for users to use for setting logLevel. The file will look as follows: 
+
+```ts
+export namespace EXAMPLE_LOG_BINDINGS {
+  export const METADATA = 'example.log.metadata';
+  export const APP_LOG_LEVEL = 'example.log.level';
+  export const TIMER = 'example.log.timer';
+  export const LOG_ACTION = 'example.log.action';
+}
+
+export enum LOG_LEVEL {
+  DEBUG,
+  INFO,
+  WARN,
+  ERROR,
+  OFF,
+}
+```
+
+### `/src/log.component.ts`
+
+Keep this file the same but updating the `Bindings` according to the values we declared in `/src/keys.ts`. This is where providers are exported so they can automatically bind to the appropriate keys. The final version of this file will look as follows:
+
+```ts
+import {EXAMPLE_LOG_BINDINGS} from './keys';
+import {Component, ProviderMap} from '@loopback/core';
+import {TimerProvider, LogActionProvider, LogLevelProvider} from './';
+
+export class LogComponent implements Component {
+  providers?: ProviderMap = {
+    [EXAMPLE_LOG_BINDINGS.TIMER]: TimerProvider,
+    [EXAMPLE_LOG_BINDINGS.LOG_ACTION]: LogActionProvider,
+    [EXAMPLE_LOG_BINDINGS.APP_LOG_LEVEL]: LogLevelProvider,
+  };
+}
+```
+
+### `/src/types.ts`
+
+Update the types slightly as shown below to make them more robust.
+
+```ts
+import {ParsedRequest, OperationArgs} from '@loopback/rest';
+
+export interface LogFn {
+  (
+    req: ParsedRequest,
+    args: OperationArgs,
+    // tslint:disable-next-line:no-any
```

Comment:
Remove tslint directives. I'm glad you ran the code to make sure it works with our lint rules, but it's not a requirement for users.